x-pack/plugin/repository-encrypted/src/test/java/org/elasticsearch/repositories/encrypted/ChainingInputStreamTests.java

```diff
@@ -0,0 +1,573 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.repositories.encrypted;
+
+import org.elasticsearch.common.Randomness;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.mockito.Mockito;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ChainingInputStreamTests extends ESTestCase {
+
+    public void testSkipAcrossComponents() throws Exception {
```

Comment:
Can we split this into multiple methods.
It actually has 2 test cases in it now (single component, multiple components) but that's not obvious from the method name, and it would be better to make it 2 explicit test cases.

Reply:
Actually, in this test case, the skip byte count is larger than the current component can handle. And there are two cases, there is a following one, or there's not. If there is, `skip` continues to skip, otherwise it returns short.

I will make two test methods.